Algorithm.Framework/Execution/StandardDeviationExecutionModel.cs

```diff
@@ -0,0 +1,217 @@
+        /// </summary>
+        /// <param name="period">Period of the standard deviation indicator</param>
+        /// <param name="deviations">The number of deviations away from the mean before submitting an order</param>
+        /// <param name="resolution">The resolution of the STD and SMA indicators</param>
+        public StandardDeviationExecutionModel(
+            int period = 60,
+            decimal deviations = 2m,
+            Resolution resolution = Resolution.Minute
+            )
+        {
+            _period = period;
+            _deviations = deviations;
+            _resolution = resolution;
+            _targetsCollection = new PortfolioTargetCollection();
+            _symbolData = new Dictionary<Symbol, SymbolData>();
+        }
+
+        /// <summary>
+        /// Executes market orders if the standard deviation of price is more than the configured number of deviations
+        /// in the favorable direction.
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <param name="targets">The portfolio targets</param>
+        public void Execute(QCAlgorithmFramework algorithm, IEnumerable<IPortfolioTarget> targets)
+        {
+            _targetsCollection.AddRange(targets);
+
+            foreach (var target in _targetsCollection)
+            {
+                var symbol = target.Symbol;
+
+                // calculate remaining quantity to be ordered
+                var unorderedQuantity = OrderSizing.GetUnorderedQuantity(algorithm, target);
+
+                // fetch our symbol data containing our STD/SMA indicators
+                SymbolData data;
+                if (!_symbolData.TryGetValue(symbol, out data))
+                {
+                    continue;
+                }
+
+                // ensure we're receiving price data before submitting orders
+                if (data.Security.Price == 0m)
+                {
+                    continue;
+                }
+
+                // check order entry conditions
+                if (data.STD.IsReady && PriceIsFavorable(data, unorderedQuantity))
+                {
+                    // get the maximum order size based on total order value
+                    var maxOrderSize = OrderSizing.Value(data.Security, MaximumOrderValue);
+                    var orderSize = Math.Min(maxOrderSize, Math.Abs(unorderedQuantity));
+
+                    // round down to even lot size
+                    orderSize -= orderSize % data.Security.SymbolProperties.LotSize;
+                    if (orderSize != 0)
+                    {
+                        algorithm.MarketOrder(symbol, Math.Sign(unorderedQuantity) * orderSize);
+                    }
+                }
+
+                // check to see if we're done with this target
+                unorderedQuantity = OrderSizing.GetUnorderedQuantity(algorithm, target);
```

Comment:
Note: we realize this could potentially never be 0 if you're trading cross pairs; as the portfolio.Quantity and cashbook[coin].Quantity will be incompatible. WIP until resolve how to model better.

Reply:
Included in that thought experiment should be consideration for what a portfolio target against a pair means without the concept of virtual positions?